app/src/main/java/com/hfad/mymessenger/CreateMessageActivity.kt

```diff
@@ -39,13 +46,17 @@ class CreateMessageActivity : AppCompatActivity() {
         dataIntent.type = "text/plain"
         //TODO: use putExtra method with the label "Intent.EXTRA_TEXT" to attach the user's message to the intent
         //  and send it to a "Chooser" window for the user to see it with the different apps
-
+    val message = msg.text.toString()
+        dataIntent.putExtra(Intent.EXTRA_TEXT, message)
+        startActivity(dataIntent)
```

Comment:
You can also use a [ShareSheet](https://developer.android.com/training/sharing/send) window, sometimes called a "chooser"